### PR TITLE
admin: add `update-extrafiles` command

### PR DESCRIPTION
**Pull request by rjbou**
This is needed with changes in #5564.
Repositories can contain opam file with no extra-files field, as opam automatically adds them. With changes in #5564, this is no more done. This command ease updating the repository to fill all extra-files fields, and permit also with the combination with the option `--hash` to choose & update used hash in a repository.

### CODE REVIEW

File: src/client/opamAdminCommand.ml [hunk 1]
```diff
@@ -331,6 +331,100 @@ let packages_with_prefixes repo_root packages =
...
+               OpamStd.Option.default [] (OpamFile.OPAM.extra_files opam)
+             in
+             let extra_files = OpamFilename.Base.Map.of_list extra_files in
+             let xfiles =
+               List.fold_left (fun extra (file, base) ->
+                   match OpamFilename.Base.Map.find_opt base extra_files with
+                   | Some hash ->
+                     let kind = OpamHash.kind hash in
+                     let xhash = compute ~kind file in
+                     if OpamHash.compare hash xhash <> 0 then
```
Comment (kit-ty-kate) [inline src/client/opamAdminCommand.ml:396]:
```suggestion
                     if not (OpamHash.equal hash xhash) then
```

**Review by kit-ty-kate — COMMENTED** [review]
Aside from this minor stylistic suggestion, I’m not sure to understand the tests. Is there no test showing the reason for the command being there in the first place? i.e. "a file exists in `files/` but no `extra-files` is defined".

I’m only seeing the "a file exists in `files` but the `extra-files` section mismatches" case (which I didn’t really realized this was adding but is really welcome too!)

On that last point, given i feel like most usage of this will be for updating hashes instead of adding missing ones, maybe the command should be called `update-extrafiles` instead of `add-extrafiles`.

**Comment by rjbou** [comment]
The test show several cases:
* a file exists in `files/` but no present in `extra-files:`
* a file exists in `files/` but `extra-files:` mismatch (for md5 & sha256)
* a file exists in `files/` and we want to update the hash (md5 -> sha256).

The name is taken from `add-hashes` command, but it can be changed.

**Review by AltGr — APPROVED** [review]

**Comment by kit-ty-kate** [comment]
> a file exists in files/ but no present in extra-files

In that case i think a more explicit test case should be added to make sure adding `extra-files` when missing entirely happens:
```suggestion
### <packages/with-xf/with-xf.1/opam>
opam-version: "2.0"
### <packages/with-xf/with-xf.1/files/missing-hash>
content
### openssl md5 packages/with-xf/with-xf.1/files/missing-hash | '.*= ' -> '' >$ XF_MD5
### opam admin add-extrafiles
### opam-cat packages/with-xf/with-xf.1/opam | "${XF_MD5}" -> "good-md5"
extra-files: ["missing-hash" "md5=good-md5"]
opam-version: "2.0"
```

**Comment by rjbou** [comment]
test added & name updated!

File: src/client/opamAdminCommand.ml (hunk 1 above)
Comment (rjbou) [reply src/client/opamAdminCommand.ml:396]: equal is already a comparison with 0. Or you propose this change for another purpose?

**Review by kit-ty-kate — APPROVED** [review]

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**